addons/main/functions/fnc_regenBleedout.sqf

```diff
@@ -0,0 +1,40 @@
+/*
+    Credit to MajorDanvers
+    Slowly regenerates bleedout timer, faster when closer to death
+    2 minutes to regen last down to second last
+    15 minutes to regen first down
+    For coeff == 1, 1 second regenerated per duration of bleedout timer
+*/
+#include "script_component.hpp"
+params ["_unit"];
+
+#define malus GVAR(bleedOutTimeMalus)
+#define bleedout GVAR(bleedoutTime)
+#define subtraction GVAR(bleedoutTimeSubtraction)
+#define coeff GVAR(bleedoutRegenCoeff)
+
+// If player has not gone down or has fully recovered, wait until they are down
+if (isNil {malus}) then {
+    [
+        {
+            !isNil {malus} && {!(_unit getVariable [QGVAR(unconscious),false])}
+        },
+        FUNC(regenBleedout), [_unit]
+    ] call CBA_fnc_waitUntilAndExecute;
```

Comment:
Not really a fan of this, potentially a player never goes down, so this runs all the time. I think a better solution would be introductions for the event when going uncon and getting revived out of uncon.

meaning a new event to listen to, and from there we can start the regeneration. So you do not need to constantly run the regeneration itself in a loop as well. You can just stop it once the regeneration is fully done and wait for the new event to be raised.

Reply:
The loop should only be running while their down timer is below the maximum. If a player never goes down or their timer fully recovers, it will stay stuck behind the `waitUntilAndExecute`. Should that still be avoided, having a potentially endless `waitUntilAndExecute`?

There's probably a slick way to directly incorporate this in the malus calculation in `setUnconscious`, so the regen is only calculated when needed.

Reply:
> There's probably a slick way to directly incorporate this in the malus calculation in setUnconscious, so the regen is only calculated when needed.

Yes. usually it is well to avoid loops if possible. The proposed idea of mine was to add a new CBA event when ever a local unit enters or leaves unconsciousness.
When leaving uncon you either died from bleed out or otherwise, or you were revived and have suffered a malus. When that happens you call the regen function which loops itself until the malus has been removed, the unit was knocked uncon again, or the unit died.